roxie/ccd/ccdfile.cpp

```diff
@@ -288,7 +289,11 @@ class CRoxieLazyFileIO : implements ILazyFileIO, implements IDelayedFile, public
                         if (isCompressed && !isKey)
                             current.setown(createCompressedFileReader(f));
                         else
+                        {
                             current.setown(f->open(IFOread));
+                            if (current && blockedIOsize)
+                                current.setown(createBlockedIO(current.getClear(), blockedIOsize));
```

Comment:
I've just realised...  this class is not thread-safe, so can't be used for index shared-random access.

I'm not sure what we do about that because we want to allow concurrent access - so a critical section would not be a good idea.

Reply:
the blocked io class isn't thread safe, but isn't jhtree when it uses this IO, guaranteeing that it only loads from it (from CDiskKeyIndex::loadNode), in a mutex protected thread?